Core/NoOpRunner.Core/Shapes/StaticShape.cs

```diff
@@ -42,28 +43,38 @@ public Platform(int centerPosX, int centerPosY, int bottomPosY, int topPosY) : b
             }
         }
 
-        private void AddShape(bool horizontal, int posX, int posY, int blockLength)
+        protected void AddShape(bool horizontal, int posX, int posY, int blockLength)
         {
-            if (horizontal) 
+            if (horizontal)
                 MapShapeX(posX, posY, blockLength, Color.Red);
             else
                 MapShapeY(posX, posY, blockLength, Color.Red);
         }
 
-        private int RandomHeight(int maxLength)
+        protected int RandomHeight(int maxLength)
         {
             int number = RandomNumber.GetInstance().Next(maxLength * -1, maxLength);
             while (number == 0)
             {
                 number = RandomNumber.GetInstance().Next(maxLength * -1, maxLength);
             }
 
-            return RandomNumber.GetInstance().Next(maxLength*-1, maxLength);
+            return RandomNumber.GetInstance().Next(maxLength * -1, maxLength);
         }
 
-        private int RandomLength(int maxLength)
+        protected int RandomLength(int maxLength)
         {
             return RandomNumber.GetInstance().Next(2, maxLength);
         }
+
+        public override bool CanOverlap(BaseShape other)
+        {
+            return false; // Not Implemented Yet
+        }
+
+        public override void OnCollision(BaseShape other)
+        {
+            // Not Implemented Yet
+        }
```

Comment:
Same as with Entity, no need for implementations here, children will override the methods (and you can already implement some of them there, i.e. for a static impassable platform, nothing can overlap it and nothing happens on collision (from the platform's perspective; If i.e. a rocket would hit a platform, the rocket's OnCollision method would be called instead. Or maybe both methods would be called, maybe the logic should go to the objects that it affects, so the logic that explodes the rocket would go to the rocket's OnCollision, the logic that damages the player would go to the player's OnCollision. We'll see later on.)

Reply:
Fixed. Left empty OnCollision functions in Impassable and Passable platforms.